### PR TITLE
Improve bottom sheet VoiceOver support

### LINKED ISSUE #6066: Order Creation: Accessibility → The popup's dismiss button should have a label
When using VoiceOver, this should at least say "Dismiss. Button" rather than just "Button". 

j

### LINKED ISSUE #6068: Order Creation: Accessibility → Popup menu items should be marked as "Button"
When using VoiceOver, these menu items should be dictated as "Button" instead of just the title and description. The "Button" further clarifies that they should select one of these to continue.

### PR DESCRIPTION
**Pull request by rachelmcr**
Closes: #6066 
Closes: #6068 

Depends on https://github.com/wordpress-mobile/WordPressUI-iOS/pull/110
Before merging this PR:
- [x] Update Podfile to use the WordPressUI 1.12.5-beta.1 release instead of the feature branch.

## Description

Improves the VoiceOver support in the bottom sheet. This bottom sheet is used in various places in the app:

* Selecting what kind of order to create (new order or simple payment)
* Selecting what kind of product to create (simple physical, simple virtual, variable, etc.)
* Sorting product list
* Adding more details when creating/editing a product
* Selecting the upload method to add a file to a downloadable product
* Providing information about the WCServices discount when purchasing a shipping label for an order

There are two improvements:

* Adds the accessibility label "Dismiss" to the button that dismisses the bottom sheet.
* Marks options in the bottom sheet lists as buttons. (This applies to all the bottom sheets noted above except the last informational bottom sheet.)

## Changes

* Updates the `WordPressUI` pod to get the "Dismiss" accessibility label for the dismiss button in all bottom sheets.
* Adds the `button` accessibility trait to cells in `BottomSheetListSelectorViewController`.

## Testing

Enable VoiceOver on your device (or use the Accessibility Inspector in a simulator).
Go to the Orders tab and tap the "+" button.
In the bottom sheet that appears, confirm the grip button at the top of the sheet is labelled "Dismiss."
Confirm that each of the order types is labeled as a button.

You can repeat these steps to check each of the bottom sheets in the the app:

* Go to the Products tab and tap the "+" button.
* Select "Sort by" on the Products tab.
* Add or edit a product and select "Add more details."
* While editing a product, go to the Product Settings and enable "Downloadable Product." Then, select the "Downloadable files" section of the product details and select "Add File."
* Go to the Orders tab and select an order eligible for shipping labels. Select "Create Shipping Label," complete the shipping label form, and tap the "WooCommerce Discount" row.

## Screenshots

Before|After
-|-
image|NewOrder-Dismiss
image|NewOrder-Button

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile
Comment (Ecarrion) [inline Podfile]: I guess we should hold from merging until `1.12.5` is released right?

**Review by Ecarrion — APPROVED** [review]
Very nice touch!

File: Podfile
Comment (rachelmcr) [reply Podfile]: We can merge this once `1.12.5-beta.1` is released (after merging https://github.com/wordpress-mobile/WordPressUI-iOS/pull/110) and we update this branch to use it. Release managers will handle releasing `1.12.5` and updating the app to use the production version in the next code freeze (related: paNNhX-33-p2).

Comment (rachelmcr) [reply Podfile]: I made that update to use the released version of `1.12.5-beta.1` in [0e46d18](https://github.com/woocommerce/woocommerce-ios/pull/6482/commits/0e46d186b9c334a9fb3657ca7dd187e63694d729), so we're good to go now.

**Pull request merged**
